vulture_os/toolkit/api_parser/signalsciences_ngwaf/signalsciences_ngwaf.py

```diff
@@ -0,0 +1,226 @@
+            return None
+
+
+    # Entering in conflict with crontab on >1min collecting and have an incidence on system integrity (parser&logs OK btw) (~400k entries)
+    # May get a cpu peak while collecting on a long late time period while crontab trying to start another parser process
+    # Moreover crontab doesn't handle correctly the execution of multiple threads this resulting to an indefinitely run of execute() method
+    # def late_handler(self):
+    #     while self.frontend.last_api_call < timezone.now() - timedelta(minutes=6):
+    #         since = min(self.frontend.last_api_call, timezone.now() - timedelta(minutes=6))
+    #         to = min(timezone.now() - timedelta(minutes=5), since + timedelta(minutes=1))
+
+    #         logger.info(f"[{__parser__}]:execute: Catch up the delay from {since} to {to}", extra={'frontend': str(self.frontend)})
+
+    #         response = self.get_logs(since, to)
+    #         self.update_lock()
+
+    #         if logs := response.get("data", None):
+    #             self.write_to_file([self._format_log(log) for log in logs if log])
+    #             self.update_lock()
+
+    #             while response["next"].get("uri", None) not in [None, '']:
+    #                 response = self.__execute_query(self.API_BASE_URL + response['next']['uri'])
+    #                 self.update_lock()
+    #                 if logs := response.get("data", None):
+    #                     self.write_to_file([self._format_log(log) for log in logs if log])
+    #                     self.update_lock()
+
+    #             self.frontend.last_api_call = to
+
+
+    def execute(self):
+        try:
+            since = min(self.frontend.last_api_call, timezone.now() - timedelta(minutes=6))
+            to = min(timezone.now() - timedelta(minutes=5), since + timedelta(minutes=1))
+
+            logger.info(f"[{__parser__}]:execute: Start collecting logs from {since} to {to}", extra={'frontend': str(self.frontend)})
+
+            response = self.get_logs(since, to)
+            self.update_lock()
+
+            if logs := response.get("data", None):
+                self.write_to_file([self._format_log(log) for log in logs if log])
+                self.update_lock()
+
+                while response["next"].get("uri", None) not in [None, '']:
+                    response = self.__execute_query(self.API_BASE_URL + response['next']['uri'])
+                    self.update_lock()
+                    if logs := response.get("data", None):
+                        self.write_to_file([self._format_log(log) for log in logs if log])
+                        self.update_lock()
+
+                self.frontend.last_api_call = to
+
+            # If we're late about 24h -> move forward 1h to prevent API error
+            if self.frontend.last_api_call < timezone.now()-timedelta(hours=24):
+                self.frontend.last_api_call += timedelta(hours=1)
+
+            self.frontend.save()
```

Comment:
prefer using self.last_api_call while handling API's logic, then save definitive value in frontend at the end (`save()` is superfluous, but can be kept that way)